cdap-common/src/main/resources/cdap-default.xml

```diff
@@ -938,6 +938,13 @@
     </description>
   </property>
 
+  <property>
+    <name>hive.latest.version.for.unknown.version</name>
```

Comment:
I think it's better to call as `hive.version` and have the possible values `auto.exact` (Current behavior), `auto.latest` (try to match or use closest one that suits the cluster) or the actual version string (if the detection failed, user can take over)